### PR TITLE
Fixes the Elemental and LBANN packages to conform to the new syntax

### PR DESCRIPTION
**Pull request by bvanessen**
introduced by PR https://github.com/LLNL/spack/pull/1875

### CODE REVIEW

File: var/spack/repos/builtin/packages/lbann/package.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ class Lbann(CMakePackage):
     depends_on('elemental +openmp_blas +scalapack +shared +int64 +debug', when='+debug')
     depends_on('cuda', when='+gpu')
     depends_on('mpi')
-    depends_on('opencv@2.4.13', when='+opencv')
+    depends_on('opencv@3.2.0', when='+opencv')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/lbann/package.py]: Just to confirm, is this change intended to be in this PR? It wasn't mentioned in the description so I wanted to make sure it didn't slip in accidentally.

**Comment by bvanessen** [comment]
Yes, I also updated that.

Brian C. Van Essen
vanessen1@llnl.gov
(w) 925-422-9300
(c) 925-290-5470
